cpp/src/reductions/simple.cuh

```diff
@@ -56,12 +59,51 @@ std::unique_ptr<scalar> simple_reduction(column_view const& col,
     auto it = thrust::make_transform_iterator(
       dcol->pair_begin<ElementType, true>(),
       simple_op.template get_null_replacing_element_transformer<ResultType>());
-    result = detail::reduce(it, col.size(), Op{}, stream, mr);
+    result = detail::reduce(it, col.size(), simple_op, stream, mr);
   } else {
     auto it = thrust::make_transform_iterator(
       dcol->begin<ElementType>(), simple_op.template get_element_transformer<ResultType>());
-    result = detail::reduce(it, col.size(), Op{}, stream, mr);
+    result = detail::reduce(it, col.size(), simple_op, stream, mr);
   }
+
+  // set scalar is valid
+  result->set_valid((col.null_count() < col.size()), stream);
```

Comment:
What is expected if col is of size  0

Reply:
Column size of 0 will not get this far.

Reply:
https://github.com/rapidsai/cudf/pull/6666/files#diff-b71d2e78efe5329e987f3561ed4641832a6c2b85732ac782a79153cac45d3018L120